core/opl/shovel.py

```diff
@@ -58,15 +69,41 @@ def args(parser, group_actions):
             default="redhat-appstudio-load-test/artifacts/load-tests.json",
             help="Path to the artifact",
         )
+        group.add_argument("--prow-data-file", help="prow data jdon file")
 
 
 class pluginOpenSearch:
     def __init__(self, args):
         print("Hello from pluginOpenSearch init")
         self.logger = logging.getLogger("opl.showel.pluginOpenSearch")
 
-    def upload(self):
+    def upload(self, args):
         print("Hello from pluginOpenSearch upload")
+        if args.data_file is None:
+            raise Exception("A Data file is needed to work with --opensearch-upload")
+        elif args.end_timestamp is None:
+            raise Exception("End timestamp is needed to work with --opensearch-upload")
+        else:
+            json_data = json.dumps(
+                {"query": {"match": {"endTimestamp": args.end_timestamp}}}
```

Comment:
Make the field configurable via options. For some project, we might want to rely on different field.